spi/src/main/java/org/opensearch/jobscheduler/spi/utils/LockService.java

```diff
@@ -80,7 +84,10 @@ void createLockIndex(ActionListener<Boolean> listener) {
         if (lockIndexExist()) {
             listener.onResponse(true);
         } else {
-            final CreateIndexRequest request = new CreateIndexRequest(LOCK_INDEX_NAME).mapping(lockMapping());
+            // Temporarily force DOCUMENT replication until SEGMENT supports GET by id
+            // https://github.com/opensearch-project/OpenSearch/issues/8536
+            Settings replicationSettings = Settings.builder().put(SETTING_REPLICATION_TYPE, DOCUMENT.name()).build();
```

Comment:
This may be wrong here and overwriting existing settings including number of replicas.